### PR TITLE
Serialize updated at without to s

### PR DESCRIPTION
**Pull request by briandunn**
The project I'm on has an overridden ::Time::DATE_FORMATS[:default] that does not include the seconds in the serialized time string. This broke batch detection in a way that was very difficult to track down. This PR uses [iso8601](https://api.rubyonrails.org/classes/ActiveSupport/TimeWithZone.html#method-i-iso8601) To serialize reliably.

### CODE REVIEW

**Comment by briandunn** [comment]
@MaicolBen Here is a passing version.

**Comment by jeremy-ebler-vineti** [comment]
`#to_s` is an alias of [to_formatted_s](https://apidock.com/rails/Time/to_formatted_s), which interacts with `Time::DATE_FORMATS` and Rails' I18n. It makes sense to explicitly use a standardized format, like iso8601.

File: app/models/devise_token_auth/concerns/tokens_serialization.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
 module DeviseTokenAuth::Concerns::TokensSerialization
+  extend self
```
Comment (MaicolBen) [inline app/models/devise_token_auth/concerns/tokens_serialization.rb]: 👏

File: app/models/devise_token_auth/concerns/tokens_serialization.rb [hunk 2]
```diff
@@ -16,4 +18,16 @@ def self.load(json)
       json
     end
   end
+
+  private
+
+  def serialize_updated_at(token)
+    updated_at_key = ['updated_at', :updated_at].find(&token.method(:[]))
+
+    if token[updated_at_key].respond_to?(:iso8601)
```
Comment (MaicolBen) [inline app/models/devise_token_auth/concerns/tokens_serialization.rb]: maybe we can write this better:
```ruby
return token unless token[updated_at_key].respond_to?(:iso8601)

token.merge updated_at_key => token[updated_at_key].iso8601

```

File: test/models/concerns/tokens_serialization_test.rb [hunk 3]
```diff
@@ -55,16 +57,48 @@
         assert_equal(ts.dump({}), '{}')
       end
 
-      it 'deserialize tokens' do
```
Comment (MaicolBen) [inline test/models/concerns/tokens_serialization_test.rb]: why did you remove this spec? are we having breaking changes here?

File: gemfiles/rails_5_1_mongoid_7.gemfile [hunk 4]
```diff
@@ -34,7 +34,7 @@ end
 
 group :test do
   gem "rails-controller-testing"
-  gem "simplecov", require: false
+  gem "simplecov", "< 0.18", require: false
```
Comment (MaicolBen) [inline gemfiles/rails_5_1_mongoid_7.gemfile]: I don't know why this change but it's ok

Comment (bdvineti) [reply gemfiles/rails_5_1_mongoid_7.gemfile]: Newer simplecov is incompatible with codeclimate. https://github.com/codeclimate/test-reporter/issues/413

**Comment by MaicolBen** [comment]
I rebased it in #1423
